papis/paths.py

```diff
@@ -199,80 +201,27 @@ def get_document_hash_folder(
-    and a potentially random *seed* value. If given, the contents of the files
-    in *paths* are also added to the hash (the first *file_read_limit* bytes).
-
-    If not given, the *seed* value is random, so the folder name will be
-    different on each call to this function. If this is not desired, the seed
-    can be set to the empty string instead.
-
-    :param doc: a document from which to generate a folder name.
-    :param paths: a list of document files. If not provided, *doc* is searched
-        for a *files* key and the documents inside are used as a default.
-    :param file_read_limit: number of bytes to read from each file in *paths*,
-        if it exists. This can be set to 0 to omit adding file contents to the
-        hash.
-    :param seed: a string to add to the hash of the document.
-
-    :returns: a (potentially non-deterministic) folder name based on the
-        contents of *doc* and *paths* of the form ``HASH[-authors]``.
-    """
-    import hashlib
-    import random
-
-    if paths is None:
-        from papis.document import Document
-
-        if isinstance(doc, Document):
-            paths = doc.get_files()
-        else:
-            paths = doc.get("files", [])
-
-    if seed is None:
-        seed = str(random.random())
-
-    # NOTE: these are sorted to ensure deterministic results if seed is given
-    docdata = sorted(doc.items())
-    paths = sorted(paths)
-
-    md5 = hashlib.md5()
-    md5.update(seed.encode())
-    md5.update(str(docdata).encode())
-
-    if paths:
-        md5.update("".join(paths).encode())
-
-    if file_read_limit > 0:
-        for path in paths:
-            if not os.path.exists(path):
-                continue
-
-            with open(path, "rb") as fd:
-                md5.update(fd.read(file_read_limit))
+    warn("'get_document_hash_folder' is deprecated and will be removed"
+         "Use 'papis.id.compute_an_id' instead.",
+         DeprecationWarning, stacklevel=2)
 
-    author = "-{:.20}".format(doc["author"]) if "author" in doc else ""
-    return normalize_path(f"{md5.hexdigest()}{author}")
+    from papis.id import compute_an_id
+    return compute_an_id(from_data(dict(doc)), seed)
 
 
 def get_document_folder(
         doc: DocumentLike,
-        dirname: PathLike,
-        in_document_paths: Iterable[str], *,
+        dirname: PathLike, *,
         folder_name_format: Optional[str] = None) -> str:
     """Generate a folder name for the document at *dirname*.
 
     This function uses :confval:`add-folder-name` to generate a folder name for
     the *doc* at *dirname*. If no folder can be constructed from the format, then
-    :func:`get_document_hash_folder` is used instead. The returned folder name
-    is guaranteed to be unique and to be a subfolder of *dirname*.
+    the reference's papis_id is used instead as a subfolder of *dirname*. The
+    papis_id is guaranteed to be unique.
```

Comment:
```suggestion
    the document's ``papis_id` is used instead as a subfolder of *dirname*. The
    ``papis_id`` is guaranteed to be unique.
```

Reply:
I looked up the asymmetric backticks. I assume they're a typo?

Reply:
Yep, that was a typo!